acra-censor/acra-censor_configuration_provider.go

```diff
@@ -36,24 +41,40 @@ const (
 
 // Config shows handlers configuration: queries, tables, patterns
 type Config struct {
-	Handlers []struct {
+	Version          string `yaml:"version"`
+	IgnoreParseError bool   `yaml:"ignore_parse_error"`
+	ParseErrorsLog   string `yaml:"parse_errors_log"`
+	Handlers         []struct {
 		Handler  string
 		Queries  []string
 		Tables   []string
 		Patterns []string
 		FilePath string
 	}
-	IgnoreParseError bool   `yaml:"ignore_parse_error"`
-	ParseErrorsLog   string `yaml:"parse_errors_log"`
 }
 
+// ErrUnsupportedConfigVersion acra-censor's config has version less than MinimalCensorConfigVersion
+var ErrUnsupportedConfigVersion = errors.New("acra-censor's config is outdated")
+
 // LoadConfiguration loads configuration of AcraCensor
 func (acraCensor *AcraCensor) LoadConfiguration(configuration []byte) error {
 	var censorConfiguration Config
 	err := yaml.Unmarshal(configuration, &censorConfiguration)
 	if err != nil {
 		return err
 	}
+	configVersion, err := utils.ParseVersion(censorConfiguration.Version)
```

Comment:
current censor configs don't have version, so it can be parsed. shouldn't we `return `ErrUnsupportedConfigVersion`?

Reply:
it will return [this](https://github.com/Lagovas/acra/blob/lagovas/T971-censor-versioning/acra-censor/acra-censor_configuration_provider.go#L57) error and log will looks like:
```
[...] "Can't setup censor" error="VERSION value has incorrect format (semver 2.0.0 format expected, https://semver.org/)"
```
There return another error value to distinguish cases when version is missed/has incorrect value and version is outdated. Maybe better to return 3 different error values: missed version/incorrect format/outdated?

Plus I think to change "VERSION value has incorrect format (semver 2.0.0 format expected, https://semver.org/)" -> "version value has incorrect format (semver 2.0.0 format expected, https://semver.org/)". Because at start it was a reference to variable VERSION

Reply:
added a check on empty version and return `
ErrUnsupportedConfigVersion`

Reply:
thank you, makes sense!